generics/helpers/common_handler_v2.js

```diff
@@ -0,0 +1,719 @@
+											}
+										}
+									})
+								}
+							})
+							.catch((err) => {
+								resolve(err)
+							})
+					}
+				})
+			})
+		} catch (err) {
+			resolve(err)
+		}
+	})
+}
+
+async function ganttChartObject(projects) {
+	return new Promise(async function (resolve, reject) {
+		let arrayOfChartData = []
+		let projectData = []
+		let i = 1
+
+		await Promise.all(
+			projects.map(async (eachProject) => {
+				let data = []
+				let labels = []
+				let leastStartDate = ''
+
+				await Promise.all(
+					eachProject.projectDetails[0].tasks.map((eachTask) => {
+						if (eachTask.startDate) {
+							leastStartDate = eachTask.startDate
+						}
+						labels.push(eachTask.title)
+						data.push({
+							task: eachTask.title,
+							startDate: eachTask.startDate,
+							endDate: eachTask.endDate,
+						})
+					})
+				)
+				if (data.length > 0) {
+					data.forEach((v) => {
+						leastStartDate = new Date(v.startDate) < new Date(leastStartDate) ? v.startDate : leastStartDate
+					})
+				}
+
+				let chartOptions = {
+					order: 1,
+					options: {
+						type: 'horizontalBar',
+						data: {
+							labels: labels,
+							datasets: [
+								{
+									data: data.map((t) => {
+										if (leastStartDate && t.startDate) {
+											return dateDiffInDays(new Date(leastStartDate), new Date(t.startDate))
+										}
+										return null
+									}),
+									datalabels: {
+										color: '#025ced',
+										//   formatter: function (value, context) {
+										//     return '';
+										//   },
+									},
+									backgroundColor: 'rgba(63,103,126,0)',
+									hoverBackgroundColor: 'rgba(50,90,100,0)',
+								},
+								{
+									data: data.map((t) => {
+										if (t.startDate && t.endDate) {
+											return dateDiffInDays(new Date(t.startDate), new Date(t.endDate))
+										}
+									}),
+									datalabels: {
+										color: '#025ced',
+										//   formatter: function (value, context) {
+										//     return '';
+										//   },
+									},
+								},
+							],
+						},
+						options: {
+							maintainAspectRatio: false,
+							title: {
+								display: true,
+								text: eachProject.projectDetails[0].title,
+							},
+							legend: { display: false },
+							scales: {
+								x: {
+									stacked: true,
+									ticks: {
+										callback: function (value, index, values) {
+											if (leastStartDate) {
+												const date = new Date(leastStartDate)
+												date.setDate(value)
+												return getDate(date)
+											}
+										},
+									},
+								},
+
+								y: {
+									stacked: true,
+								},
+							},
+						},
+					},
+				}
+				arrayOfChartData.push(chartOptions)
+				eachProject.order = i
+				projectData.push(eachProject)
+				i++
+			})
+		)
+
+		resolve([arrayOfChartData, projectData])
+	})
+}
+
+function dateDiffInDays(a, b) {
+	const utc1 = Date.UTC(a.getFullYear(), a.getMonth(), a.getDate())
```

Comment:
@MallanagoudaB this is a common function, check if similar function is present in utils , if not move it to utils. Do not use a,b values for params. Use meaning full variable name